src/components/booksform.js

```diff
@@ -1,32 +1,87 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import Proptypes from 'prop-types';
+import { createAction } from '../actions/index';
+import randomNumber from '../logic';
 
-const BooksForm = () => {
-  const bookCategories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
-  return (
-    <form>
-      <fieldset className="form-group">
-        <label htmlFor="formInput">
-          title
-          <input
-            type="text"
-            className="form-control"
-            placeholder="Enter a book Title"
-            id="formInput"
-          />
-        </label>
-      </fieldset>
-      <fieldset>
-        <select>
-          {bookCategories.map(category => (
-            <option key={category} value={category}>
-              {category}
-            </option>
-          ))}
-        </select>
-      </fieldset>
-      <button type="submit">Submit</button>
-    </form>
-  );
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e) {
+    if (e.target.id === 'formInput') {
+      this.setState({ title: e.target.value });
+    } else {
+      this.setState({ category: e.target.value });
+    }
+  }
```

Comment:
### [ optional ]

this can also be written as 
```
```javascript
handleChange(event) {
  const name = event.target.name;

  this.setState({
    title: event.target.value
    [name]: event.target.value
  })
}
```
```